lib/pg_eventstore/queries.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require_relative 'query_builders/events_filtering_query'
+
+module PgEventstore
+  class Queries
```

Comment:
I think this class could be split. I mean we can have very specialized classes lets discuss it. 
I dont need serializer deserializer for example for the create_stream or find_stream

Reply:
Well, yes. You can split it. But then you will have to inject each instance of such classes in your commands classes. For example, Append command uses `#transaction`, `#find_or_create_stream`, `#insert` and `#update_stream_revision` queries.

- `#transaction` doesn't do any reads or writes
- `#find_or_create_stream` does reads and writes
- `#insert` insert does writes only
- `#update_stream_revision` does updates

Should those 4 methods be a part of 4 different classes?


Currently, `Queries` class is focused on providing the implementation to query PostgreSQL. It already falls under Single Responsibility principle, so it should be just fine

Reply:
Not sure what you mean @aroeczek but i got confused about the class name, because its called "Queries". A query usually is a read. But in terms of db query it can also be writes, so it is okay. But maybe it should rather be named `Operations` or `API`?

Reply:
"Queries" states here for "PostgreSQL queries"